src/score_hv/harvesters/soca_diags.py

```diff
@@ -0,0 +1,301 @@
+                which variable to parse SOCA output.
+        Methods:
+        --------
+        get_data: parse descriptive statistics from SOCA output.
+    
+                  returns a list of tuples containing specific data
+    """
+    config: SOCADiagsConfig = field(default_factory=SOCADiagsConfig)
+     
+    def get_data(self):
+        depths = self.config.depths
+        groups_wanted = ['ObsValue','oman','ombg']
+        units = None
+        for filename in self.config.harvest_filenames:
+            harvested_data = list()
+            filename_info = parse_filename(filename)
+            print("Opening file: ",filename)
+            try:
+               dataset = netCDF4.Dataset(filename, 'r')
+            except Exception as e: 
+               raise OSError (f"Failed to open NetCDF file {filename}: {e}")
+
+            print("file is open ",filename)
+            """
+              The information about the file from the 
+              file name.
+              """
+            variable = filename_info['variable_type']  
+            filetime_str = filename_info['datetime']
+            dt_obj = dt.strptime(filetime_str, "%Y%m%d%H")
+            filetime = dt_obj.strftime("%Y-%m-%d %H:%M:%S")
+            sensor = filename_info['sensor']
+            file_region = filename_info['region']
+            satellite = filename_info['satellite']
+            level = filename_info['level']
+            latitude,longitude = read_MetaData_group(dataset)
+            
+            """
+              We can now loop through the list of wanted groups.
+              We get the values from the wanted groups and the FillValue.
+              """
+            for group in groups_wanted:
+                if group not in dataset.groups:
+                   print(group," is not in the dataset.groups ",dataset.groups)
+                   sys.exit(1)
+                else:     
+                   requested_group = dataset.groups[group]
+                   group_name = group
+                   num_variables = len(requested_group.variables)
+                   for var_name in requested_group.variables:
+                       variable = var_name
+                       var = requested_group.variables[var_name]
+                       if num_variables > 1:
+                          longname = var_name 
+                       else:
+                          longname = list(requested_group.variables.keys())[0] 
+
+                       if "units" in var.ncattrs():
+                          units = str(var.getncattr("units"))
+   
+                       groupvalue_variable = requested_group.variables[var_name]
+                       var_values = np.array(groupvalue_variable[:])
+
+                       if '_FillValue' in groupvalue_variable.ncattrs():
+                          fill_value = groupvalue_variable.getncattr('_FillValue')
+
+                       if np.ma.isMaskedArray(var_values):
```

Comment:
I'm not sure why this if/else block that modifies var_values is necessary. Is it for safety purposes for mixed formats of the input data wrt how missing data are treated?